exec/exec_test.go

```diff
@@ -101,3 +103,14 @@ func TestExecutableNotFound(t *testing.T) {
 		t.Errorf("Expected error ErrExecutableNotFound but got %v", err)
 	}
 }
+
+func TestTimeout(t *testing.T) {
+	exec := New()
+	ctx, cancel := context.WithTimeout(context.Background(), time.Nanosecond)
+	defer cancel()
+
+	err := exec.CommandContext(ctx, "sleep", "2").Run()
+	if err == nil {
+		t.Errorf("expected error but got %v", err)
+	}
```

Comment:
Is there any deeper assertion we can do to be sure the error is what we expetced?